pandas/tests/io/test_excel.py

```diff
@@ -1150,109 +1148,114 @@ def test_excel_sheet_by_name_raise(self, *_):
         with pytest.raises(xlrd.XLRDError):
             pd.read_excel(xl, "0")
 
-    def test_excel_writer_context_manager(self, *_):
+    def test_excel_writer_context_manager(self, frame, frame2, *_):
         with ExcelWriter(self.path) as writer:
-            self.frame.to_excel(writer, "Data1")
-            self.frame2.to_excel(writer, "Data2")
+            frame.to_excel(writer, "Data1")
+            frame2.to_excel(writer, "Data2")
 
         with ExcelFile(self.path) as reader:
             found_df = pd.read_excel(reader, "Data1", index_col=0)
             found_df2 = pd.read_excel(reader, "Data2", index_col=0)
 
-            tm.assert_frame_equal(found_df, self.frame)
-            tm.assert_frame_equal(found_df2, self.frame2)
+            tm.assert_frame_equal(found_df, frame)
+            tm.assert_frame_equal(found_df2, frame2)
 
-    def test_roundtrip(self, merge_cells, engine, ext):
-        self.frame['A'][:5] = nan
+    def test_roundtrip(self, merge_cells, engine, ext, frame):
+        frame = frame.copy()
```

Comment:
Note that any assignments to `self.frame` had to be replaced with a `copy` call to avoid SettingWithCopy warnings

Reply:
there's float_string_frame and a float_frame_with_na in pandas\tests\frame\conftest.py which could be promoted to pandas\conftest.py and avoid some of this. (just a suggestion. take a look. see what you think)

Reply:
Given how convoluted the first review process was might make sense to keep fixture consolidation and movement outside of this module to a follow up (or pre-cursor)

Reply:
I would do a follow-up.  It'll be easier to consolidate when the dust settles.